### PR TITLE
[UI] Fix bug for RunLinksPopover

### PR DESCRIPTION
**Pull request by harupy**
## What changes are proposed in this pull request?

The run links popover unintentionally shows up on relayout by double click (please take a look at the attached gif). This PR fixes it.

demo

## How is this patch tested?

Manually tested

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s) does this PR affect?

- [ ] UI
- [ ] CLI
- [ ] API
- [ ] REST-API
- [ ] Examples
- [ ] Docs
- [ ] Tracking
- [ ] Projects
- [ ] Artifacts
- [ ] Models
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/server/js/src/components/MetricsPlotPanel.js [hunk 1]
```diff
@@ -260,6 +260,7 @@ export class MetricsPlotPanel extends React.Component {
    * and schema.
    */
   handleLayoutChange = (newLayout) => {
+    this.isClicked = false;
```
Comment (harupy) [inline mlflow/server/js/src/components/MetricsPlotPanel.js]: I haven't yet figured out what causes this bug, but setting `isClicked` to `false` in `handleLayoutChange` fixes the bug.

**Review by smurching — COMMENTED** [review]
@harupy thanks for the fix! After some investigation, this particular fix seems to depend on Plotly invoking the `handleLayoutChange` callback before the `updatePopover` callback - in practice, this seems to always be the case, but it'd be good to think if there's a more robust solution that would work if the callback ordering were reversed.

In particular, it seems the following occurs (after putting `debugger`s in `handleLayoutChange` and `updatePopover`) when the plot is double-clicked to reset the layout
* The plot range is reset (i.e. Plotly resets its own plot range - this change is visible in the UI)
* `updatePopover` is called. At this point, the plot range has already updated, so it seems Plotly mistakenly believed a point was clicked & has hence called `updatePopover`. We set `this.isClicked = !this.isClicked`, which in this case sets `this.isClicked` to `true`, then trigger a timeout that will update the popover state accordingly in 300ms
* `handleLayoutChange` is called and sets `this.isClicked` to false
* The timeout created in `updatePopover` runs, observes that `this.isClicked` is false, and closes the popover

**Comment by smurching** [comment]
If a more robust solution isn't possible, I don't think it needs to block this PR, would just be good to spend some time investigating

**Comment by harupy** [comment]
@smurching Thanks for the investigation!

I checked when plotly calls `onClick`.

- When the plot is double-clicked, `onClick` is called twice.
- The first `onClick` is called **before** the relayout.
- The second `onClick` is called **after** the relayout.
- `onClick` is called only when a point is selected.

If the first click is performed with a point selected, and the second one without, `onClick` is called only once,  which sets `isClicked` to `true`, then the popover shows up.

**Comment by harupy** [comment]
Summarized when the popover shows up.

| point selected (1st click) | point selected (2nd click) | popover shows up? |
| :------------------------- | :------------------------- | :---------------- |
| yes                        | no                         | yes               |
| no                         | yes                        | yes               |
| yes                        | yes                        | no                |
| no                         | no                         | no                |

**Comment by harupy** [comment]
I also investigated the execution order of `onClick`, `onDoubleClick`, and `onRelayout`. The second `onClick` is called after `onDoubleClick`. To prevent the popover from showing up, we need to set `isClicked` in `onRelayout` (which is `handleLayoutChange`).

order-callbacks

**Review by smurching — COMMENTED** [review]
Got it, yeah sounds good to merge this fix as is - maybe we can rename `isPlotClicked` to `displayPopover` for clarity? Otherwise LGTM

**Comment by harupy** [comment]
`displayPopover` sounds better. Thanks!

**Review by smurching — APPROVED** [review]
LGTM

**Comment by harupy** [comment]
@smurching tests passed

**Pull request merged**